### PR TITLE
Don't percent-encode non-ASCII characters in fragments

### PR DESCRIPTION
**Pull request by swankjesse**
Fixes https://github.com/square/okhttp/issues/1635

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 1]
```diff
@@ -774,14 +774,15 @@ public Builder encodedPath(String encodedPath) {
...
+          ? queryStringToNamesAndValues(canonicalize(query, QUERY_ENCODE_SET, false, true, false))
           : null;
       return this;
     }
 
     public Builder encodedQuery(String encodedQuery) {
       this.encodedQueryNamesAndValues = encodedQuery != null
-          ? queryStringToNamesAndValues(canonicalize(encodedQuery, QUERY_ENCODE_SET, true, true))
+          ? queryStringToNamesAndValues(
+          canonicalize(encodedQuery, QUERY_ENCODE_SET, true, true, false))
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Shouldn't this be indented more?

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 2]
```diff
@@ -1537,21 +1540,23 @@ static int decodeHexDigit(char c) {
    *
    * @param alreadyEncoded true to leave '%' as-is; false to convert it to '%25'.
    * @param query true if to encode ' ' as '+', and '+' as "%2B".
+   * @param fragment true to leave non-ASCII codepoints as-is; false to encode them.
    */
   static String canonicalize(String input, int pos, int limit, String encodeSet,
-      boolean alreadyEncoded, boolean query) {
+      boolean alreadyEncoded, boolean query, boolean fragment) {
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: This describes the consumption location, not the actual behavior. This method knows nothing about fragments. Maybe call it `encodeToAscii`? or `retainNonAscii`? I don't know... naming is hard.

Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: (Granted it's in a file that only concerns itself with urls)

**Comment by JakeWharton** [comment]
lgtm

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java (hunk 1 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: It's not how IntelliJ does it. Fixed manually.

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java (hunk 2 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: I inverted the boolean & used `asciiOnly`. That's semantic.

**Pull request merged**